documentation/book/con-resource-kafka.adoc

```diff
@@ -0,0 +1,801 @@
+
+`selector` (optional)::
+Allows to select a specific persistent volume to use. It contains a `matchLabels` field which contains key:value pairs representing labels for selecting such a volume.
+
+`delete-claim` (optional)::
+Boolean value which specifies if the persistent volume claim has to be deleted when the cluster is undeployed. The default value is `false`.
+
+.Example fragment of a `Kafka` resource configuring Kafka with `persistent-storage` and 1Gi `size`
+[source,json]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 1Gi
+    # ...
+----
+
+The following example demonstrates use of a storage class.
+
+.Example fragment of a `Kafka` resource configuring Kafka with `persistent-storage` using a storage class
+[source,json]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 1Gi
+      class: my-storage-class
+    # ...
+----
+
+Lastly, a `selector` can be used in order to select a specific labelled persistent volume which provides some needed features like an SSD.
+
+.Example fragment of a `Kafka` resource configuring Kafka with _match labels_ selector
+[source,json]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 1Gi
+      selector:
+        matchLabels:
+          "hdd-type": "ssd"
+      deleteClaim: true
+    # ...
+----
+
+When the _persistent-claim_ is used, other than the resources already described in the xref:kafka-config-map-details-{context}[Kafka] section, the following resources are generated :
+
+`data-[cluster-name]-kafka-[idx]`::
+Persistent Volume Claim for the volume used for storing data for the Kafka broker pod `[idx]`.
+
+`data-[cluster-name]-zookeeper-[idx]`::
+Persistent Volume Claim for the volume used for storing data for the Zookeeper node pod `[idx]`.
+
+See xref:type-EphemeralStorage[`EphemeralStorage` type v1alpha1 kafka.strimzi.io] and xref:type-PersistentClaimStorage[`PersistentClaimStorage` type v1alpha1 kafka.strimzi.io] for more information.
+
+[id='metrics-{context}]
+== Metrics
+
+{ProductName} uses the link:https://github.com/prometheus/jmx_exporter[Prometheus JMX exporter^] in order to expose metrics on each node.
+It is possible to configure a `metrics` object in the `kafka` and `zookeeper` objects in `Kafka` resources, and likewise a `metrics` object in the `spec` of `KafkaConnect` resources.
+In all cases the `metrics` object should be the configuration for the JMX exporter.
+You can find more information on how to use it in the corresponding GitHub repo.
+
+For more information about using the metrics with Prometheus and Grafana, see xref:metrics[Metrics]
```

Comment:
@scholzj , the link mentioned here is leading to this section itself. Should I be linking it to some other section or website?

Reply:
I think it should be linking to the Appendix D: http://strimzi.io/docs/master/#metrics-str